server/tracedb/datastoreresource/repository.go

```diff
@@ -0,0 +1,174 @@
+package datastoreresource
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/kubeshop/tracetest/server/id"
+	"github.com/kubeshop/tracetest/server/resourcemanager"
+)
+
+var Operations = []resourcemanager.Operation{
+	resourcemanager.OperationGet,
+	resourcemanager.OperationUpdate,
+}
+
+func NewRepository(db *sql.DB) *Repository {
+	return &Repository{db}
+}
+
+type Repository struct {
+	db *sql.DB
+}
+
+func (r *Repository) SetID(dataStore DataStore, id id.ID) DataStore {
+	dataStore.ID = id
+	return dataStore
+}
+
+const insertQuery = `
+INSERT INTO data_stores (
+	"id",
+	"name",
+	"type",
+	"is_default",
+	"values",
+	"created_at"
+) VALUES ($1, $2, $3, $4, $5, $6)`
+
+const deleteQuery = `DELETE FROM data_stores`
+
+func (r *Repository) getCreatedAt(ctx context.Context, dataStore DataStore) (string, error) {
+	if dataStore.CreatedAt != "" {
+		// client passed date, keeping it
+		return dataStore.CreatedAt, nil
+	}
+
+	// get datastore on the database or the default one
+	oldDataStore, err := r.Get(ctx, dataStore.ID)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			// record not found, return a new date
+			return time.Now().UTC().Format(time.RFC3339Nano), nil
+		}
+
+		return "", err
+	}
+
+	// record found, return old date
+	return oldDataStore.CreatedAt, nil
+}
+
+func (r *Repository) Update(ctx context.Context, dataStore DataStore) (DataStore, error) {
+	// enforce ID and default
+	dataStore.ID = "current"
+	dataStore.Default = true
+
+	// reuse the created_at field for auditing purposes,
+	// unless the client explicitly send it
+	createdAt, err := r.getCreatedAt(ctx, dataStore)
```

Comment:
do we need to support this case? I think unless we have an explicit need for this, clients should not be allowed to set this

Reply:
We have this case today, I believe that we can remove it on the future.